buildSrc/src/main/kotlin/scripts/variants.gradle.kts

```diff
@@ -0,0 +1,46 @@
+package scripts
+
+plugins { id("core.android") }
+
+private enum class BuildTypes(val value: String) {
+    DEBUG("debug"), RELEASE("release")
+}
+
+private enum class ProductFlavors(val value: String) {
+    DEV("dev"), INTERNAL("internal"), PUBLIC("public")
+}
+
+private enum class FlavorDimensions(val value: String) {
```

Comment:
@Gardosen do you have a suggestion? Please also suggest how you would do it instead. That is what we are expecting from code reviews :). Otherwise that would be a complain. :)

A version is a version here in this case, I do not see any other way of describing it. :thinking:

Reply:
FlavorDimensions is a gradle thing, so I think the name is ok. However, the name "version" is a bit confusing for me. Why did we choose that name? How does versioning contribute to dimensions?

Reply:
Dimension is use to group flavors. You can group flavors based on different contexts/dimensions. In this case I did not find a better way of grouping this. Suggestions?

If we were to have for example these flavors:
 - Premium
 - Pro
 - Semi-Pro

Then dimension would be payments, because they are based/belong to this area.

Reply:
I usually put "default" as the name, if I have a single dimension

Reply:
```default``` does not tell me much. And I'm also considering to extend this with other dimensions related to "customer" which are custom builds.

Reply:
Flavor Dimensions allow us to define groups of flavors. Gradle creates every combination between these groups for us. No need to mix and match by hand. That is why I find ```default``` a bit senseless. In any case, I want to merge this so I will get back to it. It is already modified.